impl/asm/src/main/java/org/wildfly/transformer/asm/TransformerImpl.java

```diff
@@ -58,10 +69,10 @@
     private static final boolean useASM7 = getMajorJavaVersion() >= 11;
     private boolean classTransformed;
     private boolean alreadyTransformed;
-    private boolean alreadyAddedClassForNameMethod;
     private String changeClassName;
     final Map<String, String> mappingWithSeps;
     final Map<String, String> mappingWithDots;
+    final Set<String> generatedReflectionModelHandlingCode = new HashSet<>();
```

Comment:
Since Transformer can be used by multiple threads, 
HashSet should be CopyOnWriteArraySet or other synchronization technique should be used.

Reply:
+1 for multiple thread support, I wasn't thinking about that, but your right, we should do that still.

Reply:
updated.